core/ir/instr_api.h

```diff
@@ -287,7 +287,10 @@ struct _instr_t {
     /* translation target for this instr */
     app_pc translation;
 
-    uint opcode;
+    uint opcode : 16;
+
+    /* category of this instr (e.g. branch, load/store, etc.) */
+    uint category : 16;
```

Comment:
This does seem like a good solution for your initial category set: I think shrinking the opcode will not cause any problems.  However, after some discussions here about categories we'd like to add on top of your initial ones, we're thinking a cap of 16 categories is too limiting.  We're about to bump the major version from 9 to 10 anyway so we're thinking let's go ahead and have a separate 32-bit category field and break compatibility now while the version bump is happening.

I'm seeing 3 32-bit fields at the top: if you add `uint category` after the `length` field it will not affect the 64-bit size (b/c it must currently have a padding slot to align the pointer field `bytes`) which will stay 112. 

This is still a compatibility break for 32-bit -- for that, update the black box as you did before, and the heap.c comment (I assume order didn't change there: asserts will fire if so), and should really bump the between-releases version too to satisfy our version policy https://dynamorio.org/page_new_release.html via: replace "9.93" with "9.94" in these files:
```
$ git grep '9\.93'
.github/workflows/ci-docs.yml:          export VERSION_NUMBER=9.93.$((`git log -n 1 --format=%ct` / (60*60*24)))
.github/workflows/ci-package.yml:          export VERSION_NUMBER=9.93.$((`git log -n 1 --format=%ct` / (60*60*24)))
.github/workflows/ci-package.yml:          export VERSION_NUMBER=9.93.$((`git log -n 1 --format=%ct` / (60*60*24)))
.github/workflows/ci-package.yml:          export VERSION_NUMBER=9.93.$((`git log -n 1 --format=%ct` / (60*60*24)))
.github/workflows/ci-package.yml:          export VERSION_NUMBER=9.93.$((`git log -n 1 --format=%ct` / (60*60*24)))
.github/workflows/ci-package.yml:          export VERSION_NUMBER=9.93.$((`git log -n 1 --format=%ct` / (60*60*24)))
.github/workflows/ci-package.yml:          export VERSION_NUMBER="9.93.$((`git log -n 1 --format=%ct` / (60*60*24)))"
CMakeLists.txt:set(VERSION_NUMBER_DEFAULT "9.93.${VERSION_NUMBER_PATCHLEVEL}")
```
Plus change `OLDEST_COMPATIBLE_VERSION_DEFAULT` to "994" in the top-level CMakeLists.txt.
(Or if PR #5835 goes in first and already bumps these to 9.94: bump further to 9.95.)

(Note that even with this change as-is to make `opcode` a 16-bit field we have compatibility breakage as the upper bits of opcode will now be set, even if no size change.)

Reply:
Done. Upgrade to 9.95 and add to changes list